cfme/utils/template/base.py

```diff
@@ -184,13 +186,37 @@ def ssh_client_args(self):
                 'username': credentials[cred_key]['username'],
                 'password': credentials[cred_key]['password']}
 
+    @cached_property
+    def raw_vm_ssh_client_args(self):
+        """ Returns credentials + hostname for ssh client auth for a temp_vm."""
+
+        def check_ip():
+            vm_name = (self.temp_vm_name if self.provider_data.type == 'rhevm' else
```

Comment:
Might be able to remove this confusion altogether by making a settable property for the temp VM, and setting it during the upload run() method. So after a template deploy, set `self._temp_vm_mgmt` to the vm object before returning True and moving on to further VM config.